src/main/java/com/shipmentEvents/demo/ContactApp.java

```diff
@@ -0,0 +1,57 @@
+package com.company;
+
+import java.io.FileWriter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+public class ContactApp {
+
+    private static Map<String, Contact> contacts = new HashMap<>();
+
+    public static void main(String[] args) throws Exception {
+
+        Scanner in = new Scanner(System.in);
+
+        System.out.println("Please enter the contact name:");
+        String name = in.nextLine();
+
+        System.out.println("Please enter the contact phone number:");
+        String number = in.nextLine();
+
+        Contact person1 = new Contact();
+        person1.setName(name);
+        person1.setPhoneNumber(number);
+
+        // add contact to map
+        if(!contacts.containsKey(person1.getName())) {
+            contacts.put(person1.getName(), person1);
+        }
+
+        System.out.println("Please enter the contact name:");
+        name = in.nextLine();
+
+        System.out.println("Please enter the contact phone number:");
+        number = in.nextLine();
+
+        Contact person2 = new Contact();
+        person2.setName(name);
+        person2.setPhoneNumber(number);
+
+        // add contact to map
+        if(!contacts.containsKey(person2.getName())) {
+            contacts.put(person2.getName(), person2);
+        }
+
+        writeToFile();
+    }
+
+
+    public static void writeToFile() throws Exception {
+        try {
+            FileWriter myWriter = new FileWriter("contacts.txt");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *myWriter*. The resource is referenced at line 53. The resource is closed at line 54. There are other execution paths that do not contain closure statements, for example, when Writer.write throws an exception. Ensure *myWriter* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).